### PR TITLE
math: rewrite hypot(), update tests

### PR DESCRIPTION
**Pull request by tankf33der**
Reworked my tests, added more candidates to the edge cases list and `hypot()` got caught in the net. 
Didn't find anything better than rewriting the function from [golang](https://github.com/golang/go/blob/master/src/math/hypot.go).
Added one test that panics on `master`.

The patch passes all my tests against `mpfr`.

### CODE REVIEW

File: vlib/math/hypot.v [hunk 1]
```diff
@@ -1,25 +1,27 @@
 module math
 
-// hypot returns the hypotenuse of the triangle give two sides
+// hypot returns the hypotenuse of the triangle give two sides.
```
Comment (spytheman) [inline vlib/math/hypot.v:3]:
```suggestion
// hypot returns the hypotenuse of a triangle given two of its sides.
```
